xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/renderer/DefaultPageAttachmentURITitleGenerator.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.internal.renderer;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.rendering.listener.reference.ResourceReference;
+import org.xwiki.rendering.renderer.reference.link.URITitleGenerator;
+
+/**
+ * Generate link titles for ATTACH URIs.
+ *
+ * @version $Id$
+ * @since 15.2-RC1
+ */
+@Component
+@Named("doc")
+@Singleton
+public class DefaultPageAttachmentURITitleGenerator implements URITitleGenerator
+{
+    @Override
+    public String generateCreateTitle(ResourceReference reference)
+    {
+        return "Create page: " + reference.getReference();
```

Comment:
There should be a translation used here no?

Reply:
This is the default behavior, with the rendering module on its own. AFAIK there is no access to translations here. 
See [the full implementation in xwiki-platform](https://github.com/xwiki/xwiki-platform/blob/79fc276ba085daaa56c3efad5aca94265b9ca8b8/xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/java/org/xwiki/rendering/internal/renderer/XWikiPageAttachmentURITitleGenerator.java#L60).

Reply:
In this case, we should discuss a way to allow xwiki platform to inject the a translation here. Having some text in an unknown language is not going to help accessibility for non-English speaking users.

Reply:
This behavior is overwritten by the xwiki platform class I linked above.
I followed the pattern used so far when we needed translations in xwiki-rendering.
See Michael's comment on the [jira issue](https://jira.xwiki.org/browse/XWIKI-19383).

Other similar uses of this pattern can be found [here](https://github.com/xwiki/xwiki-rendering/blob/22eac2b1d38fbe259fa72e45df30f78eab553325/xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/renderer/DefaultPageAttachmentURILabelGenerator.java#L44) and [here](https://github.com/xwiki/xwiki-rendering/blob/bda9e53453ddabbe2d58490c313707f2c7a11dde/xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/renderer/DataURILabelGenerator.java#L44).
I thought it would be better to have, in all cases, a readable english title instead of just a value that could hold anything (possibly complete gibberish). We could remove this default formatting to just return reference.getReference() but I don't think it'd be best.